src/vs/workbench/contrib/inlineChat/browser/inlineChatController.ts

```diff
@@ -146,6 +146,7 @@ export class InlineChatController implements IEditorContribution {
 	}
 
 	dispose(): void {
+		this._strategy?.dispose();
```

Comment:
I might be wrong but I don't believe this is needed, `finishExistingSession` should wind down everything which should also dispose the current startegy

Reply:
It appears this is needed, the leak detector pings if I remove the line. Maybe this is a bug in the test or elsewhere in the controller? Feel free to push changes!

Reply:
Makes sense. The tests don't really await the (state machine's) run to terminate and I'd say it makes them to ugly to enforce that.